BrainPortal/app/views/tasks/show.html.erb

```diff
@@ -57,6 +57,26 @@
       <% confirm += "\n\nIMPORTANT NOTE: This will also remove its archived work directory." if @task.workdir_archive %>
       <%= link_to 'Remove Task',  { :action => 'operation', :operation => 'delete', 'tasklist[]' => @task.id }, :confirm => confirm, :class  => "button menu_button", :method => :post %>
     <% end %>
+    <% if @task.status =~ /Failed*|Completed/ %>
+      <%= link_to "Remove Work Directory", url_for(:action  => :operation, :operation => 'zap_wd', 'tasklist[]' => @task.id),
+                                                  :method  => :post,
+                                                  :datatype => :script,
+                                                  :class => "button",
+                                                  :confirm =>
+                  "Are you sure you want to remove the task's work directory?"
+      %>
+    <% end %>
+
+    <!-- ARCHIVING -->
+    <% if @task.status =~ /Completed/ && @task.full_cluster_workdir.blank? == false && @task.archived_status.nil? %>
```

Comment:
Don't compare with `thing.blank? == false`, use `thing.present?`, it's much more legible.